web/src/main/java/com/bobocode/bring/web/actuator/service/impl/DefaultActuatorService.java

```diff
@@ -0,0 +1,102 @@
+package com.bobocode.bring.web.actuator.service.impl;
+
+import com.bobocode.bring.core.annotation.Service;
+import com.bobocode.bring.web.actuator.dto.Commit;
+import com.bobocode.bring.web.actuator.dto.GitInfo;
+import com.bobocode.bring.web.actuator.dto.PropertiesData;
+import com.bobocode.bring.web.actuator.service.ActuatorService;
+import com.bobocode.bring.web.server.properties.ServerProperties;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.InputStream;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+/**
+ * Default implementation of the {@link ActuatorService} interface.
+ * Provides methods for preparing Git information and system properties data for the Actuator endpoint.
+ *
+ * @author Blyzhnytsia Team
+ * @since 1.0
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class DefaultActuatorService implements ActuatorService {
+
+    // Constant for error messages
+    private static final String GIT_PROPERTIES_FILE_NOT_FOUND_MESSAGE = "File '%s' not found";
+
+    // The name of the git file
+    private static final String GIT_FILE_NAME = "git.properties";
+
+    // Keys of properties of git
+    private static final String GIT_BRANCH_PROPERTY = "git.branch";
+    private static final String GIT_COMMIT_ID_PROPERTY = "git.commit.id";
+    private static final String GIT_COMMIT_MESSAGE_FULL_PROPERTY = "git.commit.message.full";
+    private static final String GIT_COMMIT_TIME_PROPERTY = "git.commit.time";
+
+    // ServerProperties for configuration
+    private final ServerProperties serverProperties;
+
+    /**
+     * Prepares Git information for the Actuator endpoint.
+     *
+     * @return Git information including branch details and the latest commit information.
+     */
+    @Override
+    public GitInfo prepareGitInfo() {
```

Comment:
We could add a new method with @PostConstruct and call it one time and after that just use result this method